apiclient/src/qs_api/mod.rs

```diff
@@ -77,43 +79,34 @@ impl ApiClient {
         request_params: QsRequestParamsOut,
         token_or_signing_key: AuthenticationMethod<'_, T>,
     ) -> Result<QsProcessResponseIn, QsRequestError> {
-        let tbs = ClientToQsMessageTbsOut::new(request_params);
-        let message = match token_or_signing_key {
-            AuthenticationMethod::Token(token) => ClientToQsMessageOut::from_token(tbs, token),
-            AuthenticationMethod::SigningKey(signing_key) => tbs
-                .sign(signing_key)
-                .map_err(|_| QsRequestError::LibraryError)?,
-            AuthenticationMethod::None => ClientToQsMessageOut::without_signature(tbs),
+        let api_version = self.negotiated_versions().qs_api_version();
+
+        let request_params =
+            QsVersionedRequestParamsOut::with_version(request_params, api_version)?;
+        let message = sign_params(request_params, &token_or_signing_key)?;
+
+        let endpoint = self.build_url(Protocol::Http, ENDPOINT_QS);
+        let response = send_qs_message(&self.client, &endpoint, &message).await?;
+
+        // check if we need to negotiate a new API version
+        let Some(accepted_versions) = extract_api_version_negotiation(&response) else {
+            return process_response(response).await;
         };
```

Comment:
Sorry if I'm exessively nitpicky, but I still find the semantics of `extract_api_version_negotiation` a bit unintuitive with it returning an `Option`, where `None` is the happy path. Can't we just inline it to the point where we have something like
```rust
    if response.status() != StatusCode::NOT_ACCEPTABLE {
        return process_response(response);
    };
    let accepted_versions = parse_accepted_versions_header(response.headers())?;
```
(With `parse_accepted_versions_header` returning an error if it can't actually find the header.)

Or is your concern that we're going to have that same code in the other APIs?